crates/worker/src/partition/mod.rs

```diff
@@ -220,6 +228,7 @@ pub struct PartitionProcessor<Codec, InvokerSender, T> {
     leadership_state: LeadershipState<InvokerSender, T>,
     state_machine: StateMachine<Codec>,
     bifrost: Bifrost,
+    snapshot_producer: SnapshotProducer,
```

Comment:
From what I can see, `SnapshotProducer` is created per processor and owned by it, I'm not sure I see the value of having it passed down as an input (it's also not generic). Is there something I don't see here?

Reply:
I'm not quite following - you might have to expand a bit? Are you saying we might be better off constructing the `SnapshotProducer` in the Partition Processor itself? I'm still fumbling with what's the best way to wire things up. One improvement I can see is to introduce a `CreateSnapshot` trait and hoist either the current sync method, or a new "spawn background task to create snapshot" method, and have the PP only depend on that behavior.